### DISCUSSION TITLE
[Feature Request]: Cannot use Xdebug when using BetterPHPUnit in VSCode

### DISCUSSION THREAD

**Discussion by joshhanley** [beyondcode/herd-community#395] [Ideas]
### Operating system version

macOS Sonoma 14.1.1

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.4.0 (Build: 20)

### PHP Version

PHP 8.2.15 (cli) (built: Jan 19 2024 00:15:08) (NTS)

### Bug description

Loving all the new Herd updates. Thanks to the team for all their great work! 😁

If you use an extension like BetterPHPUnit in VSCode, or even run a CLI tool like Pest, there is no way to use Xdebug as both use the symlinked PHP version which doesn't include the Xdebug config which gets dynamically added when using `herd debug`.

I know I can manually copy the config from `Herd/config/php/XX/debug/debug.ini` and paste it into the main `Herd/config/php/XX/php.ini` but I prefer to keep the config in separate files if possible.

It would be nice if there was a way to tell Herd to just include the Xdebug config by default for all requests. Or at least tell it to load all configs from a `Herd/config/php/XX/config/` directory, that way we could move the `debug.ini` into the directory if we want it to run all the time.

### Steps to reproduce

Set up Xdebug in VSCode and add a breakpoint in a Pest test.

Run test either using BetterPHPUnit extension or on the command line using `vendor/bin/pest`.

Breakpoint isn't triggered.

If you run `php --ini` you can see that the `Herd/config/php/XX/debug/debug.ini` config isn't loaded.

It is specified in the documentation to use Xdebug in the CLI, to run `herd debug` but that isn't possible from tools that use PHP shell script type hint.

**Comment by sschlein** [comment]
We'll discuss this as a feature request internally but it sounds like that makes sense, for example adding a "Use Xdebug permanently" option to this settings screen instead of changing php.ini files permanently.

![CleanShot 2024-02-13 at 10 20 31](https://github.com/beyondcode/herd-community/assets/2911113/1a5fefe6-e02e-42ea-8a8e-cb3e3249764f)

**Reply by joshhanley** [reply]
@sschlein ok great thanks! Yep that sounds good.

I did run into an issue with my testing yesterday whereby Xdebug throws an error when the debugging client isn't running:

```
Xdebug: [Step Debug] Could not connect to debugging client. Tried: localhost:9003 (fallback through xdebug.client_host/xdebug.client_port).
```

So I don't know if having Xdebug running all the time will be suitable. Maybe the button could go into the menu bar instead?

<img width="245" alt="image" src="https://github.com/beyondcode/herd-community/assets/882837/0224ea52-cb28-4b04-82c8-e0fad28e8de1">

I'd imaging underneath "Mails" having an "Xdebug" with the green/grey dot to denote whether xdebug config is on or off and maybe making it clickable so it can be easily toggled on or off. Or something like that anyway.

**Comment by morrislaptop** [comment]
I tried changing the pest binary in the extension settings which seems like it should work, but I got the 

<img width="561" alt="image" src="https://github.com/beyondcode/herd-community/assets/67807/72437387-b464-4f93-af09-5a738e5bfa3d">


```
 Executing task: herd debug vendor/bin/pest /Users/craigmorris/Dev/pcl-stripe-cb-playground/app/Models/Order/Insurance/InsuranceTest.php --filter 'it generates the right amounts for deposit' 

Xdebug: [Step Debug] Could not connect to debugging client. Tried: localhost:9003 (through xdebug.client_host/xdebug.client_port).
```

In fact if I just run `herd debug artisan` I get similar connection errors. Is herd debug supposed to start up xdebug before executing the command? If so it doesn't seem to be working!

**Reply by joshhanley** [reply]
@morrislaptop you will still need to ensure your IDE is actively listening for an xdebug call. So you still need to manually start that (except for PHPStorm, Herd can start it for you).

**Reply by joshhanley** [reply]
@morrislaptop I also have added this to my `php.ini` to silence the warning when the debugger in the IDE isn't running

```bash
; Use this so it doesn't complain when the server is not running
xdebug.log_level=0
```

It does mean though that if there are any xdebug internal errors, you won't see any of them.

**Reply by morrislaptop** [reply]
Ah that's it! Pest debugging now super easy!

**Reply by joshhanley** [reply]
Nice! 😁
